### PR TITLE
[CI] Refactor CI

### PR DESCRIPTION
**Pull request by wangxiyuan**
1. remove some useless test func and file
2. fix format.sh problem
3. enable full test for singlecard and multicard
4. move long term test to long_term folder. For this kind of test, it only runs by labeled and daily test. Include: spec decode、accuracy test

## After refactor:
There are 4 test modules
  - `singlecard`: contains the test running on one NPU. It'll be run for each PR and daily test.
  - `multicard`: contains the test running on multi NPUs. It'll be run for each PR and daily test.
  - `long_term`: contains the test that cost much time(Now include `spec decode` and `accuracy` test). It'll be run for the PR with `long-term-test` labeled and daily test.
  - `e2e`: contains the test for doc and pd feature. It'll be run for the PR with `pd-test` labeled and daily test.

## Todo:
1. some test are skipped, they should be fixed and reenabled in the future.
2. pyhccl test for multicard doesn't work at all. It should be enabled as well.
3. ensure long-term-test pass by daily test.

### Know issue
Now, `ready-for-test` labels is required to start pd test or long term test. And when `long-term-test` or `pd-test` is labeled after another one, the old labeled test will be re-run again. So the labeled test should be ran in the following step:

1. decide which test need run, then label it. `long-term-test` or `pd-test` or both.
2. add `ready-for-test` label, then the test will be ran.

### CODE REVIEW

File: .github/workflows/vllm_ascend_test.yaml [hunk 1]
```diff
@@ -112,58 +113,36 @@ jobs:
         run: |
           if [[ "${{ matrix.os }}" == "linux-arm64-npu-1" ]]; then
             pytest -sv tests/singlecard/test_offline_inference.py
-            pytest -sv tests/singlecard/test_ilama_lora.py
-            pytest -sv tests/ops
-            pytest -sv tests/compile
+            # AscendScheduler doesn't work, fix it later
```
Comment (wangxiyuan) [inline .github/workflows/vllm_ascend_test.yaml:111]: this should be fixed by @zzzzwwjj  in https://github.com/vllm-project/vllm-ascend/pull/939

File: .github/workflows/vllm_ascend_test.yaml [hunk 2]
```diff
@@ -112,58 +113,36 @@ jobs:
...
+            # pytest -sv tests/singlecard/tets_schedule.py
+            # guided decoding doesn't work, fix it later
```
Comment (wangxiyuan) [inline .github/workflows/vllm_ascend_test.yaml:113]: this should be fixed by @shen-shanshan in https://github.com/vllm-project/vllm-ascend/pull/969

File: tests/singlecard/test_camem.py [hunk 3]
```diff
@@ -59,9 +58,9 @@ def test_basic_camem():
     assert torch.allclose(output, torch.ones_like(output) * 3)
 
 
-@fork_new_process_for_each_test
+@pytest.mark.skipif(os.getenv("VLLM_USE_V1") == "1",
```
Comment (wangxiyuan) [inline tests/singlecard/test_camem.py]: this should be fixed by @Potabk

File: tests/singlecard/sample/test_rejection_sampler.py [hunk 4]
```diff
@@ -322,6 +322,7 @@ def test_deterministic_when_seeded(
                 assert torch.equal(results[j][i], results[0][i])
 
 
+@pytest.mark.skipif(True, reason="Test failed, need fix")
```
Comment (wangxiyuan) [inline tests/singlecard/sample/test_rejection_sampler.py:325]: this should be fixed by @ponix-j

File: tests/singlecard/test_camem.py [hunk 5]
```diff
@@ -59,9 +57,9 @@ def test_basic_camem():
     assert torch.allclose(output, torch.ones_like(output) * 3)
 
 
+@pytest.mark.skipif(True, reason="test failed, should be fixed later")
```
Comment (wangxiyuan) [inline tests/singlecard/test_camem.py:60]: This should be fixed by @Potabk

File: .github/workflows/vllm_ascend_test.yaml [hunk 6]
```diff
@@ -112,58 +113,28 @@ jobs:
...
 
       - name: Run vllm-project/vllm-ascend test on V0 engine
         env:
           VLLM_USE_V1: 0
         run: |
           if [[ "${{ matrix.os }}" == "linux-arm64-npu-1" ]]; then
-            pytest -sv tests/singlecard/test_ilama_lora.py
             pytest -sv tests/singlecard/test_offline_inference.py
-            pytest -sv tests/ops
+            # AscendScheduler doesn't work, fix it later
```
Comment (wangxiyuan) [inline .github/workflows/vllm_ascend_test.yaml:127]: ditto @zzzzwwjj

File: .github/workflows/vllm_ascend_test.yaml [hunk 7]
```diff
@@ -112,58 +113,28 @@ jobs:
...
+            # pytest -sv tests/singlecard/tets_schedule.py
+            # guided decoding doesn't work, fix it later
```
Comment (wangxiyuan) [inline .github/workflows/vllm_ascend_test.yaml:129]: ditto @shen-shanshan

File: tests/multicard/test_offline_inference_distributed.py [hunk 8]
```diff
@@ -45,14 +40,28 @@ def test_models_distributed(model: str,
...
-            distributed_executor_backend=distributed_executor_backend,
+            distributed_executor_backend="mp",
     ) as vllm_model:
         vllm_model.generate_greedy(example_prompts, max_tokens)
 
 
-if __name__ == "__main__":
-    import pytest
-    pytest.main([__file__])
+@pytest.mark.skipif(os.getenv("VLLM_USE_V1") == "1",
```
Comment (wangxiyuan) [inline tests/multicard/test_offline_inference_distributed.py:50]: This should be fixed by @MengqingCao when v1 is ready

File: .github/workflows/vllm_ascend_test_long_term.yaml [hunk 9]
```diff
@@ -0,0 +1,97 @@
...
+  run:
+    shell: bash -el {0}
+
+concurrency:
+  group: pr-${{ github.event.pull_request.number }}
+  cancel-in-progress: true
+
+jobs:
+  long-term-test:
+    if: ${{ contains(github.event.pull_request.labels.*.name, 'long-term-test')  && contains(github.event.pull_request.labels.*.name, 'ready') || github.event_name == 'schedule' }}
```
Comment (Yikun) [inline .github/workflows/vllm_ascend_test_long_term.yaml]: Add a note like in followup:  long-term-test will be triggered when tag 'long-term-test' & 'ready' or schedule job

File: .github/workflows/vllm_ascend_test_long_term.yaml [hunk 10]
```diff
@@ -0,0 +1,97 @@
...
+        run: |
+          pip install -r requirements-dev.txt
+          pip install -v -e .
+
+      - name: Run vllm-project/vllm-ascend long term test
+        run: |
+          # spec decode test
+          VLLM_USE_MODELSCOPE=true pytest -sv tests/long_term/spec_decode/e2e/test_v1_spec_decode.py
+          VLLM_USE_MODELSCOPE=True pytest -sv tests/long_term/spec_decode/e2e/test_mtp_correctness.py  # it needs a clean process
+          pytest -sv tests/long_term/spec_decode --ignore=tests/long_term/spec_decode/e2e/test_mtp_correctness.py --ignore=tests/long_term/spec_decode/e2e/test_v1_spec_decode.py
```
Comment (Yikun) [inline .github/workflows/vllm_ascend_test_long_term.yaml:98]: also cc @mengwei805 the spec decode will be triggered by manaully after this PR.

Comment (wangxiyuan) [reply .github/workflows/vllm_ascend_test_long_term.yaml:98]: See the commit message:
```
1. decide which test need run, then label it. It can be `long-term-test` or `pd-test` or both.
2. add `ready-for-test` label, then the test will be ran.
```

Comment (mengwei805) [reply .github/workflows/vllm_ascend_test_long_term.yaml:98]: i consider it is a wonderful refactor
